LibertyDSNP/example-client
# Chore/connect sdk

Purpose
---------------
Up until now, the example client has been using mock services to build UX.


Solution
---------------
This PR adds preliminary support for connecting to the chain to Post and retrieve Posts. This is a bit sketchy at the moment (no tests, some hard-coded data), but it's a good point to check in an start making incremental improvements.

Change summary
---------------
* Integrate the @dsnp/sdk
* Add an example static server that permits uploads
* Add Store adapter that works with the static server.
* Add SDK code that will set SDK config.
* Add SDK method to an activity pub subscription, retrieve and parse batch files, retrieve activity content and add it as FeedItems.
* Add SDK method to post a feed item, that stores the content in the static server, adds its announcement to a batch, and saves it to the static server, and then posts the publication to the chain.


Steps to Verify
----------------
1. Start a test chain on 8545 with the lastest @dsnp/contracts deployed.
2. `cd static-server; npm install; npm run start`
3. In another terminal, npm run start.
4. Install Metamask if you have not already.
5. In the example client, hit the log in button.
6. Connect to the chain using an private key pair that has funds.
7. Click the "Post" button, fill in the form and submit. You should be taken to a Metamask window.
8. Confirm the transaction.
9. Wait 10-20 seconds for the post to land on the chain. You should see it in the feed section.
